sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -131,7 +131,7 @@ private[sql] case class JSONRelation(
 
   override def equals(other: Any): Boolean = other match {
     case that: JSONRelation =>
-      (this.path == that.path) && (this.schema == that.schema)
+      (this.path == that.path) && (DataType.equalsIgnoreNullability(this.schema, that.schema))
```

Comment:
Since nullability affects how parquet encode and decode data, I think it is better to leave `ParquetRelation2.equals` unchanged for now.